portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,90 +14,80 @@
-            
-            Task task = new Task(id, timestamp, comment);
-            tasks.add(task);
-        }
-
-        Gson gson = new Gson();
-
-        response.setContentType("application/json;");
-        response.getWriter().println(gson.toJson(tasks));
-    }
+public class DataServlet extends HttpServlet {
+  // This method takes comments, timestamps and the id of comments and creates a Task object.
+  // It then converts the Task object to a json.
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
 
-    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException 
-    {
-        String comment = getParameter(request, "text-input", "");
-        long timestamp = System.currentTimeMillis();
+    DatastoreService savedComments = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = savedComments.prepare(query);
 
-        Entity taskEntity = new Entity("Task");
-        taskEntity.setProperty("comment", comment);
-        taskEntity.setProperty("timestamp", timestamp);
+    List<Task> tasks = new ArrayList<>();
 
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        datastore.put(taskEntity);
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      long timestamp = (long) entity.getProperty("timestamp");
+      String comment = (String) entity.getProperty("comment");
 
-        response.sendRedirect("/index.html");
+      Task task = new Task(id, timestamp, comment);
+      tasks.add(task);
     }
 
-    private String getParameter(HttpServletRequest request, String name, String defaultValue) 
-    {
-        String value = request.getParameter(name);
-        
-        if (value == null) 
-        {
-        return defaultValue;
-        }
+    Gson gson = new Gson();
 
-        return value;
-    }
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(tasks));
+  }
+
+  // This method takes input from the comment box and stores it with the rest of the comments
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = getParameter(request, "text-input", "");
+    long timestamp = System.currentTimeMillis();
+
+    Entity taskEntity = new Entity("Task");
+    taskEntity.setProperty("comment", comment);
+    taskEntity.setProperty("timestamp", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(taskEntity);
+
+    response.sendRedirect("/index.html");
+  }
+
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    value = (value == null) ? defaultValue : value;
+
+    return value;
```

Comment:
nit: can simplify these two lines to:

`
return (value == null) ? defaultValue : value;
`

if you find that more readable.